### PR TITLE
[PotentialFlow] Upgrade move modelpart process using transformation and parallel utilities

### PR DESCRIPTION
**Pull request by marcnunezc**
**Description**
So far this process was coded to work on 2D only. With these changes, 3D rotations in any orientation are also contemplated.

Uses rotational and translational matrices from core. 

**Changelog**
- Using geometrical transformation utilities from core
- Replace OpenMP block with parallel utilies
- Added rotation axis as a new setting

### CODE REVIEW

File: applications/CompressiblePotentialFlowApplication/custom_processes/move_model_part_process.cpp [hunk 1]
```diff
@@ -45,31 +48,45 @@ MoveModelPartProcess::MoveModelPartProcess(ModelPart& rModelPart,
...
 
-    #pragma omp parallel for
-    for(int i = 0; i <  static_cast<int>(mrModelPart.NumberOfNodes()); ++i) {
+    Matrix translation_matrix = ZeroMatrix(4,4);
+    GeometricalTransformationUtilities::CalculateTranslationMatrix(1.0, translation_matrix, mOrigin);
+
+    Matrix rotation_matrix = ZeroMatrix(4,4);
+    GeometricalTransformationUtilities::CalculateRotationMatrix(mRotationAngle, rotation_matrix, mRotationAxis, mRotationPoint);
+
+    IndexPartition<size_t>(mrModelPart.NumberOfNodes()).for_each(
```
Comment (philbucher) [inline applications/CompressiblePotentialFlowApplication/custom_processes/move_model_part_process.cpp]: I think you can use the `block_for_each` here, the index is a leftover from OpenMP

**Comment by philbucher** [comment]
also adding a test wouldn't hurt ;)

File: applications/CompressiblePotentialFlowApplication/custom_processes/move_model_part_process.cpp (hunk 1 above)
Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/custom_processes/move_model_part_process.cpp]: True, thanks!

**Review by philbucher — APPROVED** [review]
maybe can be moved to the core eventually

**Pull request merged**